core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQConnection.scala

```diff
@@ -102,3 +104,45 @@ class DefaultRabbitMQConnection[F[_]](connection: ServerConnection,
     createChannel().use(f)
   }
 }
+
+object DefaultRabbitMQConnection {
+  def make[F[_]](connection: ServerConnection,
+                 info: RabbitMQConnectionInfo,
+                 republishStrategy: RepublishStrategyConfig,
+                 connectionListener: ConnectionListener,
+                 channelListener: ChannelListener,
+                 consumerListener: ConsumerListener,
+                 blocker: Blocker)(implicit F: ConcurrentEffect[F], timer: Timer[F], cs: ContextShift[F]): F[DefaultRabbitMQConnection[F]] =
+    F.delay {
+        new DefaultRabbitMQConnection(connection, info, republishStrategy, connectionListener, channelListener, consumerListener, blocker)
+      }
+      .flatTap { conn =>
+        conn.withChannel { channel =>
+          setUpRepublishing(republishStrategy, info, channel)
+        }
+      }
+
+  // prepare exchange for republishing
+  private def setUpRepublishing[F[_]: Sync](republishStrategyConfig: RepublishStrategyConfig,
+                                            connectionInfo: RabbitMQConnectionInfo,
+                                            channel: ServerChannel): F[Unit] = Sync[F].delay {
+    republishStrategyConfig match {
+      case RepublishStrategyConfig.CustomExchange(exchangeName, exchangeDeclare, _) =>
+        if (exchangeDeclare) {
+          DefaultRabbitMQClientFactory.declareExchange(
+            name = exchangeName,
+            `type` = ExchangeType.Direct,
+            durable = true,
+            autoDelete = false,
+            arguments = DeclareArgumentsConfig(),
+            channel = channel,
+            connectionInfo = connectionInfo
+          )
+        }
+
+        ()
+
+      case _ => () // no-op
```

Comment:
Does this mean if `RepublishStrategy` is not defined, there is no republish or there is a fallback to previous behaviour ?